lisa/tests/hotplug.py

```diff
@@ -264,4 +269,193 @@ def cpuhp_seq(cls, nr_operations, hotpluggable_cpus, max_cpus_off, random_gen):
         for cpu in cur_off_cpus:
             yield cpu, 1
 
-# vim :set tabstop=4 shiftwidth=4 textwidth=80 expandtab
+
+class HotplugRollback(FtraceTestBundle):
+
+    @classmethod
+    def _online(cls, target, cpu, online, verify=True):
+        try:
+            if online:
+                target.hotplug.online(cpu)
+            else:
+                target.hotplug.offline(cpu)
+        except TargetStableError as e:
+            if verify:
+                raise e
+
+    @classmethod
+    def _reset_fail(cls, target, cpu):
+        target.hotplug.fail(cpu, -1)
+
+    @classmethod
+    def _state_can_fail(cls, target, cpu, state, up):
+        """
+        There are no way of probing the kernel for a list of hotplug states
+        that can fail and for which we can test the rollback. We need therefore
+        to try:
+        - If we can set the state in the kernel 'fail' interface.
+        - If the hotplug is reset actually failing (some states can fail only
+          when going up or down)
+        """
+        try:
+            target.hotplug.fail(cpu, state)
+        except TargetStableError:
+            return False
+
+        try:
+            cls._online(target, cpu, up)
+            cls._reset_fail(target, cpu)
+            cls._online(target, cpu, not up)
+            """
```

Comment:
Should be a comment, not a string literal. String literals are only "replacing comments" in docstrings, as it has a special support from the Python parser (and Sphinx autodoc extends that to all class attribute and top-level definitions in modules)